execute.c

```diff
@@ -536,17 +555,30 @@ jv jq_next(jq_state *jq) {
       stack_save(jq, pc - 1, stack_get_pos(jq));
 
       stack_push(jq, jv_number(jq->subexp_nest));
+      stack_push(jq, jv_copy(jq->value_at_path));
       stack_push(jq, v);
 
       jq->path = jv_array();
+      jv_free(jq->value_at_path);
+      jq->value_at_path = jv_copy(v);
```

Comment:
Yes. The path is empty at this point, and v is the "." object on top of the stack as we start building the path. The next INDEX operation must index into v.